From ae708ca095ac6d0285d39d95818c1bb63cd01861 Mon Sep 17 00:00:00 2001
From: Claudio DeSouza <cdesouza@brave.com>
Date: Mon, 4 Nov 2024 22:12:21 +0000
Subject: [PATCH] [CodeHealth][rust] Brave `BUILD.gn` handlebars changes

This change contains the handlebars changes we apply to our own
handlebars file. This commit also adds a patch that can be used to
update the handlebar files, based on new upstream additions.
---
 .../rust/chromium_crates_io/BUILD.gn.hbs      | 37 ++++++++-----------
 1 file changed, 16 insertions(+), 21 deletions(-)

diff --git a/third_party/rust/chromium_crates_io/BUILD.gn.hbs b/third_party/rust/chromium_crates_io/BUILD.gn.hbs
index 4cf2f4f5f8b..ecf0f026241 100644
--- a/third_party/rust/chromium_crates_io/BUILD.gn.hbs
+++ b/third_party/rust/chromium_crates_io/BUILD.gn.hbs
@@ -53,14 +53,14 @@ cargo_crate("{{this.name}}") {
     ":lib",
     {{/if}}
     {{#each this.packages}}
-      "//third_party/rust/{{this.name}}/{{this.epoch}}:lib",
+      "//brave/third_party/rust/{{this.name}}/{{this.epoch}}:lib",
     {{/each}}
   ]
   {{else}}
   if ({{this.cond}}) {
     deps += [
       {{#each this.packages}}
-      "//third_party/rust/{{this.name}}/{{this.epoch}}:lib",
+      "//brave/third_party/rust/{{this.name}}/{{this.epoch}}:lib",
       {{/each}}
     ]
   }
@@ -69,7 +69,7 @@ cargo_crate("{{this.name}}") {
   {{#if aliased_deps}}
   aliased_deps = {
     {{#each aliased_deps}}
-    {{this.0}} = "{{this.1}}"
+    {{this.0}} = "//brave/third_party/rust/{{this.1}}:lib"
     {{/each}}
   }
   {{/if}}
@@ -77,14 +77,14 @@ cargo_crate("{{this.name}}") {
   {{#if @first}}
   build_deps = [
     {{#each this.packages}}
-      "//third_party/rust/{{this.name}}/{{this.epoch}}:buildrs_support",
+      "//brave/third_party/rust/{{this.name}}/{{this.epoch}}:buildrs_support",
     {{/each}}
   ]
   {{else}}
   if ({{this.cond}}) {
     build_deps += [
     {{#each this.packages}}
-      "//third_party/rust/{{this.name}}/{{this.epoch}}:buildrs_support",
+      "//brave/third_party/rust/{{this.name}}/{{this.epoch}}:buildrs_support",
     {{/each}}
     ]
   }
@@ -119,23 +119,32 @@ cargo_crate("{{this.name}}") {
   ]
   {{/with}}
   {{/if}}
-  {{#if rustenv}}
+  {{#if (or rustenv extra_kv.rustenv)}}
   rustenv = [
     {{#each rustenv}}
     "{{this}}",
     {{/each}}
+    {{#each extra_kv.rustenv}}
+    "{{this}}",
+    {{/each}}
   ]
   {{/if}}
-  {{#if rustflags}}
+  {{#if (or rustflags extra_kv.rustflags)}}
   rustflags = [
     {{#each rustflags}}
     "{{this}}",
     {{/each}}
+    {{#each extra_kv.rustflags}}
+    "{{this}}",
+    {{/each}}
   ]
   {{/if}}
   {{#if output_dir}}
   output_dir = "{{output_dir}}"
   {{/if}}
+  {{#if extra_kv.raw_gn}}
+  {{extra_kv.raw_gn}}
+  {{/if}}
   {{#if native_libs}}
   native_libs = [
   {{#each native_libs}}
@@ -144,20 +153,6 @@ cargo_crate("{{this.name}}") {
   ]
   {{/if}}
   {{/with}}
-
-  {{#if gn_visibility.public}}
-  {{#if detail.extra_kv.visibility}}
-  visibility = []
-  {{/if}}
-  {{else}}
-  # Only for usage from third-party crates. Add the crate to
-  # //third_party/rust/chromium_crates_io/Cargo.toml to use
-  # it from first-party code.
-  visibility = [ "//third_party/rust/*" ]
-  {{/if}}
-  {{#each detail.extra_kv.visibility}}
-  visibility += [ "{{this}}" ]
-  {{/each}}
   {{#if gn_visibility.testonly}}
   testonly = true
   {{/if}}
-- 
2.43.0

